src/DurableTask.Netherite/StorageLayer/Faster/AzureBlobs/AzureStorageDevice.cs

```diff
@@ -243,15 +243,31 @@ void CancelAllRequests()
                     {
                         this.BlobManager?.StorageTracer?.FasterStorageProgress($"StorageOpReturned AzureStorageDevice.WriteAsync id={id} (Canceled)");
                     }
-                    request.Callback(uint.MaxValue, request.NumBytes, request.Context);
+
+                    try
+                    {
+                        request.Callback(uint.MaxValue, request.NumBytes, request.Context);
+                    }
+                    catch (Exception ex)
+                    {
+                        this.BlobManager.StorageTracer?.FasterStorageError($"{nameof(CancelAllRequests)} for access id={id} failed during FASTER completion callback", ex);
```

Comment:
would it make sense to re-throw here? Aren't we swallowing the exception

Reply:
There is no one to catch this (it is an un-awaited task) and having exceptions in un-awaited tasks can be harmful as far as I know.

Reply:
There is no one to meaningfully catch exceptions here. And the partition has already been terminated (that's why the requests are being cancelled). So it is just about tracing at this point.